extensions/emmet/src/abbreviationActions.ts

```diff
@@ -614,14 +610,27 @@ function expandAbbr(input: ExpandAbbreviationInput): string {
 				}
 
 				// If wrapping with a block element, insert newline in the text to wrap.
-				if (wrappingNode && inlineElements.indexOf(wrappingNode.name) === -1) {
-					wrappingNode.value = '\n\t' + wrappingNode.value + '\n';
+				if (wrappingNode) {
+					wrappingNode.value = isPreviewing ? wrappingNode.value : selectedTextPlaceholder;
+					if (inlineElements.indexOf(wrappingNode.name) === -1) {
+						wrappingNode.value = '\n\t' + wrappingNode.value + '\n';
+					}
+
+					// Fixes #54711
+					if (wrappingNode.name === 'a' && !isPreviewing) {
+						wrappingNode.value = selectedTextPlaceholder;
+						let hrefAtt = wrappingNode.attributes.filter((a: any) => a.name === 'href')[0];
+						if (hrefAtt && hrefAtt.value) {
+							// Replace a *valid* with $TM_SELECTED_TEXT so that multi-cursor mode works correctly.
+							hrefAtt.value = selectedTextPlaceholder;
```

Comment:
When wrapping a url without the `http://` or `https://`, emmet outputs the http:// part in the href. Since we replace the href value here with $TM_SELECTED_TEXT, the url wont have the http://.

To see this in action look at my comment on the wrap individual lines with selection test case that you added.

Reply:
Yeah, I did wonder about that. Is it too naive to just do `href.value = "http://" + selectedTextPlaceholder;` ?

Reply:
Well, yes. Because in case of phone numbers or email, `http://` wont be the right prefix. You will have to compare `wrappingNode.value` and `hrefAtt.value` before updating the former. That will give you the right prefix to use.

Reply:
OK, updated to strip out the prefix and insert it manually now. Looks much better.